mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -956,7 +956,7 @@ private[ml] object RandomForest extends Logging {
         valueCounts.map(_._1)
       } else {
         // stride between splits
-        val stride: Double = featureSamples.length.toDouble / (numSplits + 1)
+        val stride: Double = featureSamples.size.toDouble / (numSplits + 1)
```

Comment:
This will do a second pass over the `Iterable`. Would it be preferable to combine this into the `foldLeft` above so it only does a single pass?